build failed for lib/pure/selectors via mingw

### Description

Try to cross compile the tests for [mummy](https://github.com/guzba/mummy) via `nim c -d:mingw --cpu=i386 test_http.nim` and other, which fails in `lib/pure/selectors.nim`.


### Nim Version

Nim Compiler Version 2.0.10 [Linux: amd64]
Compiled at 2024-10-06
Copyright (c) 2006-2023 by Andreas Rumpf

active boot switches: -d:release -d:danger

### Current Output

```text
/home/pulux/.cache/nim/test_http_d/@m..@s..@s..@s..@s..@susr@slib@snim@slib@spure@sselectors.nim.c: In function 'selectInto__OOZsrcZmummy_u10559':
/home/pulux/.cache/nim/test_http_d/@m..@s..@s..@s..@s..@susr@slib@snim@slib@spure@sselectors.nim.c:1184:132: error: passing argument 5 of 'select' from incompatible pointer type [-Wincompatible-pointer-types]
 1184 |         count = select((NI32)(TM__ln5vuDpFyDrAqQbYD2g52Q_26), ((fd_set*) ((&rset))), ((fd_set*) ((&wset))), ((fd_set*) ((&eset))), ptv);
      |                                                                                                                                    ^~~
      |                                                                                                                                    |
      |                                                                                                                                    tyObject_Timeval__hrHxqdi9cUhQ9cAnT79aA8JUg *
In file included from /home/pulux/.cache/nim/test_http_d/@m..@s..@s..@s..@s..@susr@slib@snim@slib@spure@sselectors.nim.c:8:
/usr/i686-w64-mingw32/include/winsock2.h:1031:116: note: expected 'const TIMEVAL *' {aka 'const struct timeval *'} but argument is of type 'tyObject_Timeval__hrHxqdi9cUhQ9cAnT79aA8JUg *'
 1031 |   WINSOCK_API_LINKAGE int WSAAPI select(int nfds,fd_set *readfds,fd_set *writefds,fd_set *exceptfds,const TIMEVAL *timeout);
      |                                                                                                     ~~~~~~~~~~~~~~~^~~~~~~
CC: ../../../../pulux/.nimble/pkgs2/crunchy-0.1.11-d34f03fc0f6876dbf53f99601096728ec31f47a3/crunchy.nim
CC: ../../../../pulux/.nimble/pkgs2/webby-0.2.1-1d9978ac16fdb56fad2ee9eaa0fcbb7860229d39/webby/internal.nim
Error: execution of an external compiler program '/usr/bin/i686-w64-mingw32-gcc -c  -w -fmax-errors=3 -mno-ms-bitfields -DWIN32_LEAN_AND_MEAN   -I/usr/lib/nim/lib -I/home/gits/nimble_gits/mummy/tests -o /home/pulux/.cache/nim/test_http_d/@m..@s..@s..@s..@s..@susr@slib@snim@slib@spure@sselectors.nim.c.o /home/pulux/.cache/nim/test_http_d/@m..@s..@s..@s..@s..@susr@slib@snim@slib@spure@sselectors.nim.c' failed with exit code: 1
```


### Expected Output

```text
Test binary output.
```


### Known Workarounds

_No response_

### Additional Information

_No response_

--- a/lib/windows/winlean.nim
+++ b/lib/windows/winlean.nim
@@ -449,14 +449,9 @@ type
 
   SockLen* = cuint
 
-when defined(cpp):
-  type
-    Timeval* {.importc: "timeval", header: "<time.h>".} = object
-      tv_sec*, tv_usec*: int32
-else:
-  type
-    Timeval* = object
-      tv_sec*, tv_usec*: int32
+type
+  Timeval* {.importc: "struct timeval", header: "<time.h>".} = object
+    tv_sec*, tv_usec*: int32
 
 var
   SOMAXCONN* {.importc, header: "winsock2.h".}: cint

